uaa/src/test/java/org/cloudfoundry/identity/uaa/mock/providers/IdentityProviderEndpointsAliasMockMvcTests.java

```diff
@@ -590,158 +739,268 @@ protected AliasFeatureDisabled() {
+                    deleteIdpViaDb(existingIdp.getOriginKey(), zone2.getId());
+
+                    // update original IdP
+                    existingIdp.setAliasId(null);
+                    existingIdp.setAliasZid(null);
+                    existingIdp.setName("some-new-name");
+                    final IdentityProvider<?> updatedIdp = updateIdp(zone1, existingIdp);
+                    assertThat(updatedIdp.getName()).isEqualTo("some-new-name");
+                    assertThat(updatedIdp.getAliasId()).isBlank();
+                    assertThat(updatedIdp.getAliasZid()).isBlank();
+                }
 
-            private void shouldReject_OnlyAliasIdSetToNull(
-                    final IdentityZone zone1,
-                    final IdentityZone zone2
-            ) throws Throwable {
-                final IdentityProvider<?> originalIdp = executeWithTemporarilyEnabledAliasFeature(
-                        aliasFeatureEnabled,
-                        () -> createIdpWithAlias(zone1, zone2)
-                );
+                @Test
+                void shouldReject_OnlyAliasIdSetToNull_UaaToCustomZone() throws Throwable {
+                    shouldReject_OnlyAliasIdSetToNull(IdentityZone.getUaa(), customZone);
+                }
 
-                assertThat(originalIdp.getAliasId()).isNotBlank();
-                assertThat(originalIdp.getAliasZid()).isNotBlank();
+                @Test
+                void shouldReject_OnlyAliasIdSetToNull_CustomToUaaZone() throws Throwable {
+                    shouldReject_OnlyAliasIdSetToNull(customZone, IdentityZone.getUaa());
+                }
 
-                originalIdp.setAliasId(null);
-                shouldRejectUpdate(zone1, originalIdp, HttpStatus.UNPROCESSABLE_ENTITY);
-            }
+                private void shouldReject_OnlyAliasIdSetToNull(
+                        final IdentityZone zone1,
+                        final IdentityZone zone2
+                ) throws Throwable {
+                    final IdentityProvider<?> originalIdp = executeWithTemporarilyEnabledAliasFeature(
+                            aliasFeatureEnabled,
+                            () -> createIdpWithAlias(zone1, zone2)
+                    );
 
-            @Test
-            void shouldReject_OnlyAliasZidSetToNull_UaaToCustomZone() throws Throwable {
-                shouldReject_OnlyAliasZidSetToNull(IdentityZone.getUaa(), customZone);
-            }
+                    assertThat(originalIdp.getAliasId()).isNotBlank();
+                    assertThat(originalIdp.getAliasZid()).isNotBlank();
 
-            @Test
-            void shouldReject_OnlyAliasZidSetToNull_CustomToUaaZone() throws Throwable {
-                shouldReject_OnlyAliasZidSetToNull(customZone, IdentityZone.getUaa());
-            }
+                    originalIdp.setAliasId(null);
+                    shouldRejectUpdate(zone1, originalIdp, HttpStatus.UNPROCESSABLE_ENTITY);
```

Comment:
Looks good. Leaving comments to track my progress as this is challenging to read through.